lib/internal/Magento/Framework/Search/Search.php

```diff
@@ -97,4 +97,40 @@ private function addFieldToFilter($field, $condition = null)
 
         return $this;
     }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return int
+     */
+    public function getTotalCount(SearchCriteriaInterface $searchCriteria)
+    {
+        $requestBuilder = \Magento\Framework\App\ObjectManager::getInstance()
+            ->get(\Magento\Framework\Search\Request\Builder::class);
+        $requestBuilder->setRequestName($searchCriteria->getRequestName());
+
+        $scope = $this->scopeResolver->getScope()->getId();
+        $requestBuilder->bindDimension('scope', $scope);
+
+        foreach ($searchCriteria->getFilterGroups() as $filterGroup) {
+            foreach ($filterGroup->getFilters() as $filter) {
+                $field = $filter->getField();
+                $condition = $filter->getValue();
+                if (!is_array($condition) || !in_array(key($condition), ['from', 'to'], true)) {
+                    $requestBuilder->bind($field, $condition);
+                } else {
+                    if (!empty($condition['from'])) {
+                        $requestBuilder->bind("{$field}.from", $condition['from']);
+                    }
+                    if (!empty($condition['to'])) {
+                        $requestBuilder->bind("{$field}.to", $condition['to']);
+                    }
+                }
+            }
+        }
+
+        $request = $requestBuilder->create();
+        $searchResponse = $this->searchEngine->search($request);
```

Comment:
the search engine should not require a second search just to figure out the correct count

Reply:
I do not find any other solution can you please suggest me a better solution to eliminate searchEngine?